hs_core/views/resource_folder_hierarchy.py

```diff
@@ -27,21 +27,21 @@ def data_store_structure(request):
     """
     res_id = request.POST.get('res_id', None)
     if res_id is None:
-        return HttpResponse(content_type='application/json', status=400)
+        return HttpResponse('Bad request - resource id is not included', status=400)

```

Comment:
@hyi  This may not be a big issue. However, I still think any data returned from the server in response to an ajax call should be in json format. The client needs to parse the Jason data and then use it as necessary. That way, in future if we need to add more data elements to response that can be easily added to the response object. 

Not sure about the following:

> HttpResponse object can be directly used without explicitly setting it, which is what has been done across our code base. 

Looking quickly at the code base for ajax response , the content_type seems to be set to 'json'.